ledger/acctupdates.go

```diff
@@ -1229,11 +1229,16 @@ func (au *accountUpdates) commitRound(offset uint64, dbRound basics.Round, lookb
 			treeTargetRound = dbRound + basics.Round(offset)
 		}
 		for i := uint64(0); i < offset; i++ {
-			err = accountsNewRound(tx, deltas[i], creatableDeltas[i], roundTotals[i+1].RewardsLevel, protos[i+1])
+			err = accountsNewRound(tx, deltas[i], creatableDeltas[i])
 			if err != nil {
 				return err
 			}
 		}
+		err = totalsNewRounds(tx, deltas[:offset], roundTotals[1:offset+1], protos[1:offset+1])
```

Comment:
Not sure I understand the change. Before `totals.applyRewards` was called `offset` times due to the loop above.
Now it is called the same amount of times due to the loop inside `totalsNewRounds`. The only difference seems like in a number of `accountsTotals` calls. So can't we call it right before the `for i := uint64(0); i < offset; i++` loop above and leave `accountsNewRound` as-is except `accountsTotals` call (not a suggestion but a question).

Reply:
Your observation is correct. Both before and after the change, `totals.applyRewards` is called `offset` times. `totals.applyRewards` is in-memory function that perform few additions on the totals data structure, and is pretty fast. On the other hand, the `accountsTotals` and `accountsPutTotals` are sql calls which aren't that fast ( relatively ), and we want to optimize these.

Reply:
i.e. taking the `totals.applyRewards` out of the loop isn't the "gain" here. The gain is by calling `accountsTotals` and `accountsPutTotals` only once.